os_net_config/cli.py

```diff
@@ -574,5 +629,33 @@ def get_iface_config(
     return iface_array
 
 
+def minimum_config(provider,
+                   min_config,
+                   no_activate,
+                   root_dir,
+                   noop):
+    if not min_config:
```

Comment:
Consider failing only if:
--provider and --purge-provider are different, no min config, and --no-network-config is not applied.